In [191]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

In [192]:
df = pd.read_csv('Airbnb_Open_Data.csv')

C:\Users\10ire\AppData\Local\Temp\ipykernel_12412\2167922700.py:1: DtypeWarning: Columns (25) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('Airbnb_Open_Data.csv')


In [193]:
df.head()

,id,NAME,host id,host_identity_verified,host name,neighbourhood group,neighbourhood,lat,long,country,...,service fee,minimum nights,number of reviews,last review,reviews per month,review rate number,calculated host listings count,availability 365,house_rules,license
0,1001254,Clean & quiet apt home by the park,80014485718,unconfirmed,Madaline,Brooklyn,Kensington,40.64749,-73.97237,United States,...,$193,10.0,9.0,10/19/2021,0.21,4.0,6.0,286.0,Clean up and treat the home the way you'd like...,NaN
1,1002102,Skylit Midtown Castle,52335172823,verified,Jenna,Manhattan,Midtown,40.75362,-73.98377,United States,...,$28,30.0,45.0,5/21/2022,0.38,4.0,2.0,228.0,Pet friendly but please confirm with me if the...,NaN
2,1002403,THE VILLAGE OF HARLEM....NEW YORK !,78829239556,NaN,Elise,Manhattan,Harlem,40.80902,-73.94190,United States,...,$124,3.0,0.0,NaN,NaN,5.0,1.0,352.0,"I encourage you to use my kitchen, cooking and...",NaN
3,1002755,NaN,85098326012,unconfirmed,Garry,Brooklyn,Clinton Hill,40.68514,-73.95976,United States,...,$74,30.0,270.0,7/5/2019,4.64,4.0,1.0,322.0,NaN,NaN
4,1003689,Entire Apt: Spacious Studio/Loft by central park,92037596077,verified,Lyndon,Manhattan,East Harlem,40.79851,-73.94399,United States,...,$41,10.0,9.0,11/19/2018,0.10,3.0,1.0,289.0,"Please no smoking in the house, porch or on th...",NaN


In [194]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 102599 entries, 0 to 102598
Data columns (total 26 columns):
 #   Column                          Non-Null Count   Dtype  
---  ------                          --------------   -----  
 0   id                              102599 non-null  int64  
 1   NAME                            102349 non-null  object 
 2   host id                         102599 non-null  int64  
 3   host_identity_verified          102310 non-null  object 
 4   host name                       102193 non-null  object 
 5   neighbourhood group             102570 non-null  object 
 6   neighbourhood                   102583 non-null  object 
 7   lat                             102591 non-null  float64
 8   long                            102591 non-null  float64
 9   country                         102067 non-null  object 
 10  country code                    102468 non-null  object 
 11  instant_bookable                102494 non-null  object 
 12  cancellation_pol

In [195]:
df.isna().sum().sort_values(ascending=False)

license                           102597
house_rules                        52131
last review                        15893
reviews per month                  15879
country                              532
availability 365                     448
minimum nights                       409
host name                            406
review rate number                   326
calculated host listings count       319
host_identity_verified               289
service fee                          273
NAME                                 250
price                                247
Construction year                    214
number of reviews                    183
country code                         131
instant_bookable                     105
cancellation_policy                   76
neighbourhood group                   29
neighbourhood                         16
long                                   8
lat                                    8
id                                     0
host id         

In [196]:
df.columns

Index(['id', 'NAME', 'host id', 'host_identity_verified', 'host name',
       'neighbourhood group', 'neighbourhood', 'lat', 'long', 'country',
       'country code', 'instant_bookable', 'cancellation_policy', 'room type',
       'Construction year', 'price', 'service fee', 'minimum nights',
       'number of reviews', 'last review', 'reviews per month',
       'review rate number', 'calculated host listings count',
       'availability 365', 'house_rules', 'license'],
      dtype='object')

In [197]:
info = pd.DataFrame({
    "col": df.columns,
    "dtype": df.dtypes.astype(str).values,
    "missing": df.isna().sum().values
})
info["missing_%"] = (info["missing"]/len(df)*100).round(2)
info.sort_values("missing_%", ascending=False)

,col,dtype,missing,missing_%
25,license,object,102597,100.00
24,house_rules,object,52131,50.81
19,last review,object,15893,15.49
20,reviews per month,float64,15879,15.48
9,country,object,532,0.52
23,availability 365,float64,448,0.44
4,host name,object,406,0.40
17,minimum nights,float64,409,0.40
21,review rate number,float64,326,0.32
22,calculated host listings count,float64,319,0.31


# Data Cleaning: Removing Columns and Rows

In this section, we remove columns and rows with excessive missing values that are not essential for our analysis.

- **Columns to drop:** `license` (100% missing) and `house_rules` (50.81% missing)
- **Rows to drop:** Records with missing values in critical columns needed for analysis

In [198]:
# Drop columns
df = df.drop(['license', 'house_rules'], axis=1)

In [199]:
# Drop rows
columns_to_check = ['availability 365','calculated host listings count','review rate number', 'country', 'host name', 'minimum nights', 'host_identity_verified','service fee','price','NAME','Construction year','number of reviews','country code','instant_bookable','cancellation_policy','neighbourhood group','neighbourhood','long','lat']
df = df.dropna(subset=columns_to_check)

# Column Name Standardization

We standardize column names to ensure consistency and ease of use throughout the analysis:

1. **Strip whitespace** from column names
2. **Convert to lowercase** for uniformity
3. **Replace underscores** with spaces for readability

In [200]:
df.columns

Index(['id', 'NAME', 'host id', 'host_identity_verified', 'host name',
       'neighbourhood group', 'neighbourhood', 'lat', 'long', 'country',
       'country code', 'instant_bookable', 'cancellation_policy', 'room type',
       'Construction year', 'price', 'service fee', 'minimum nights',
       'number of reviews', 'last review', 'reviews per month',
       'review rate number', 'calculated host listings count',
       'availability 365'],
      dtype='object')

In [201]:
df.columns = df.columns.str.strip()

In [202]:
df.columns = df.columns.str.lower()

In [203]:
df.columns = df.columns.str.replace('_', ' ')

# Handling Missing Values

We fill missing values in the dataset using appropriate imputation strategies:

- **`last review`**: Convert to datetime format and fill missing values with the median date
- **`reviews per month`**: Fill with median value after outlier analysis

In [204]:
df['last review'] = pd.to_datetime(df['last review'])
df['last review'] = df['last review'].fillna(df['last review'].median())

In [205]:
df.columns

Index(['id', 'name', 'host id', 'host identity verified', 'host name',
       'neighbourhood group', 'neighbourhood', 'lat', 'long', 'country',
       'country code', 'instant bookable', 'cancellation policy', 'room type',
       'construction year', 'price', 'service fee', 'minimum nights',
       'number of reviews', 'last review', 'reviews per month',
       'review rate number', 'calculated host listings count',
       'availability 365'],
      dtype='object')

## Outlier Detection and Imputation

The `reviews per month` column contains many outliers, as shown in the box plot below. We use the median for imputation to avoid skewing the data distribution, as the median is more robust to outliers than the mean.

In [206]:
# Box Plot
plt.figure(figsize=(10, 4))
plt.boxplot(df['reviews per month'].dropna(), vert=False)
plt.xlabel('Reviews Per Month')
plt.title('Box Plot - Outlier Check')
plt.savefig('images/reviews_per_month_outlier_boxplot.png', dpi=300, bbox_inches='tight')
plt.close()


In [207]:
df['reviews per month'] = df['reviews per month'].fillna(df['reviews per month'].median())

In [208]:
df['service fee'].unique()

array(['$193 ', '$28 ', '$41 ', '$115 ', '$212 ', '$204 ', '$58 ', '$64 ',
       '$121 ', '$116 ', '$30 ', '$219 ', '$113 ', '$74 ', '$171 ',
       '$118 ', '$168 ', '$106 ', '$13 ', '$152 ', '$105 ', '$90 ',
       '$186 ', '$66 ', '$225 ', '$19 ', '$227 ', '$32 ', '$189 ',
       '$101 ', '$23 ', '$77 ', '$190 ', '$75 ', '$231 ', '$133 ', '$35 ',
       '$82 ', '$120 ', '$21 ', '$145 ', '$65 ', '$26 ', '$89 ', '$218 ',
       '$104 ', '$44 ', '$63 ', '$79 ', '$163 ', '$22 ', '$182 ', '$165 ',
       '$37 ', '$17 ', '$48 ', '$11 ', '$131 ', '$102 ', '$240 ', '$119 ',
       '$215 ', '$191 ', '$16 ', '$233 ', '$136 ', '$170 ', '$201 ',
       '$109 ', '$123 ', '$36 ', '$12 ', '$180 ', '$238 ', '$157 ',
       '$14 ', '$198 ', '$206 ', '$122 ', '$208 ', '$140 ', '$10 ',
       '$20 ', '$110 ', '$88 ', '$98 ', '$234 ', '$151 ', '$172 ',
       '$107 ', '$138 ', '$194 ', '$139 ', '$117 ', '$31 ', '$156 ',
       '$124 ', '$175 ', '$162 ', '$187 ', '$93 ', '$51 ', '$143 ',
       '$239 '

# Data Type Conversion: Cleaning Price Fields

We clean price-related columns to enable mathematical operations:

1. Remove currency symbols (`$`)
2. Remove whitespace
3. Remove thousand separators (`,`)
4. Convert to numeric data types (`int64` for service fee, `float` for price)

This transformation allows us to perform statistical analysis and calculations on pricing data.

In [209]:
df['service fee'] = df['service fee'].str.replace('$', '')
df['service fee'] = df['service fee'].str.replace(' ', '')

df['service fee'] = df['service fee'].astype('int64')

In [210]:
df['price']

0           $966 
1           $142 
4           $204 
5           $577 
7         $1,060 
           ...   
102594      $844 
102595      $837 
102596      $988 
102597      $546 
102598    $1,032 
Name: price, Length: 99032, dtype: object

In [211]:
df['price'].unique().tolist()

['$966 ',
 '$142 ',
 '$204 ',
 '$577 ',
 '$1,060 ',
 '$1,018 ',
 '$291 ',
 '$319 ',
 '$606 ',
 '$580 ',
 '$149 ',
 '$1,097 ',
 '$1,020 ',
 '$563 ',
 '$370 ',
 '$856 ',
 '$589 ',
 '$839 ',
 '$528 ',
 '$66 ',
 '$761 ',
 '$523 ',
 '$449 ',
 '$930 ',
 '$329 ',
 '$1,125 ',
 '$96 ',
 '$1,133 ',
 '$162 ',
 '$944 ',
 '$505 ',
 '$113 ',
 '$385 ',
 '$950 ',
 '$374 ',
 '$375 ',
 '$387 ',
 '$1,155 ',
 '$665 ',
 '$175 ',
 '$408 ',
 '$601 ',
 '$107 ',
 '$724 ',
 '$450 ',
 '$327 ',
 '$131 ',
 '$444 ',
 '$1,089 ',
 '$518 ',
 '$220 ',
 '$316 ',
 '$396 ',
 '$813 ',
 '$111 ',
 '$908 ',
 '$823 ',
 '$186 ',
 '$84 ',
 '$241 ',
 '$54 ',
 '$656 ',
 '$509 ',
 '$1,198 ',
 '$1,127 ',
 '$595 ',
 '$1,076 ',
 '$956 ',
 '$81 ',
 '$1,164 ',
 '$368 ',
 '$679 ',
 '$850 ',
 '$1,003 ',
 '$544 ',
 '$614 ',
 '$178 ',
 '$61 ',
 '$901 ',
 '$1,189 ',
 '$783 ',
 '$69 ',
 '$784 ',
 '$55 ',
 '$989 ',
 '$1,029 ',
 '$787 ',
 '$725 ',
 '$612 ',
 '$824 ',
 '$1,038 ',
 '$72 ',
 '$701 ',
 '$566 ',
 '$50 ',
 '$575 ',
 '$99 ',
 '$52 ',


In [212]:
df['price'].value_counts()

price
$206       128
$481       126
$833       126
$1,056     125
$573       124
          ... 
$369        57
$923        54
$786        52
$89         52
$187        50
Name: count, Length: 1151, dtype: int64

In [213]:
df['price'] = df['price'].str.replace('$', '')
df['price'] = df['price'].str.replace(' ', '')
df['price'] = df['price'].str.replace(',', '')

df['price'] = df['price'].astype(float)

In [214]:
df['price'].unique().tolist()

[966.0,
 142.0,
 204.0,
 577.0,
 1060.0,
 1018.0,
 291.0,
 319.0,
 606.0,
 580.0,
 149.0,
 1097.0,
 1020.0,
 563.0,
 370.0,
 856.0,
 589.0,
 839.0,
 528.0,
 66.0,
 761.0,
 523.0,
 449.0,
 930.0,
 329.0,
 1125.0,
 96.0,
 1133.0,
 162.0,
 944.0,
 505.0,
 113.0,
 385.0,
 950.0,
 374.0,
 375.0,
 387.0,
 1155.0,
 665.0,
 175.0,
 408.0,
 601.0,
 107.0,
 724.0,
 450.0,
 327.0,
 131.0,
 444.0,
 1089.0,
 518.0,
 220.0,
 316.0,
 396.0,
 813.0,
 111.0,
 908.0,
 823.0,
 186.0,
 84.0,
 241.0,
 54.0,
 656.0,
 509.0,
 1198.0,
 1127.0,
 595.0,
 1076.0,
 956.0,
 81.0,
 1164.0,
 368.0,
 679.0,
 850.0,
 1003.0,
 544.0,
 614.0,
 178.0,
 61.0,
 901.0,
 1189.0,
 783.0,
 69.0,
 784.0,
 55.0,
 989.0,
 1029.0,
 787.0,
 725.0,
 612.0,
 824.0,
 1038.0,
 72.0,
 701.0,
 566.0,
 50.0,
 575.0,
 99.0,
 52.0,
 964.0,
 68.0,
 549.0,
 438.0,
 490.0,
 1137.0,
 1170.0,
 753.0,
 369.0,
 508.0,
 443.0,
 947.0,
 857.0,
 861.0,
 536.0,
 690.0,
 971.0,
 696.0,
 1188.0,
 586.0,
 608.0,
 154.0,
 132.0,
 779.0,
 619.0,
 874.0,
 8

In [215]:
df['instant bookable'].unique()
df['instant bookable'] = df['instant bookable'].astype(bool)

In [216]:
df['neighbourhood group'].unique().tolist()

['Brooklyn', 'Manhattan', 'brookln', 'Queens', 'Bronx', 'Staten Island']

In [217]:
print(df['neighbourhood group'].unique())
df['neighbourhood group']=df['neighbourhood group'].str.replace('brookln', 'Brooklyn')
df['neighbourhood group'].value_counts()

['Brooklyn' 'Manhattan' 'brookln' 'Queens' 'Bronx' 'Staten Island']


neighbourhood group
Manhattan        42252
Brooklyn         40417
Queens           12834
Bronx             2608
Staten Island      921
Name: count, dtype: int64

# Removing Duplicate Records

We identify and remove duplicate rows from the dataset to ensure data quality and prevent redundant information from affecting our analysis. This step helps maintain the integrity of our statistical calculations.

In [218]:
df = df.drop_duplicates()

In [219]:
df.shape

(98509, 24)

# Identifying Outliers and Invalid Data

We examine the distribution of key variables to identify:

- **Outliers**: Extreme values that may skew our analysis
- **Invalid data**: Impossible or incorrect values that need to be addressed
- **Data quality issues**: Inconsistencies that require cleaning

This analysis helps us understand the data quality and determine appropriate handling strategies.

In [220]:
df['availability 365'].unique().tolist()

[286.0,
 228.0,
 289.0,
 374.0,
 219.0,
 180.0,
 375.0,
 1.0,
 163.0,
 47.0,
 68.0,
 214.0,
 356.0,
 354.0,
 33.0,
 86.0,
 83.0,
 258.0,
 400.0,
 34.0,
 203.0,
 35.0,
 410.0,
 388.0,
 13.0,
 62.0,
 334.0,
 166.0,
 18.0,
 130.0,
 43.0,
 373.0,
 190.0,
 335.0,
 52.0,
 70.0,
 320.0,
 19.0,
 293.0,
 58.0,
 240.0,
 96.0,
 300.0,
 425.0,
 361.0,
 341.0,
 353.0,
 213.0,
 137.0,
 336.0,
 23.0,
 395.0,
 78.0,
 188.0,
 338.0,
 66.0,
 322.0,
 80.0,
 146.0,
 382.0,
 0.0,
 319.0,
 50.0,
 118.0,
 145.0,
 267.0,
 350.0,
 407.0,
 76.0,
 16.0,
 351.0,
 397.0,
 276.0,
 398.0,
 135.0,
 315.0,
 105.0,
 87.0,
 189.0,
 81.0,
 42.0,
 177.0,
 245.0,
 386.0,
 247.0,
 394.0,
 179.0,
 312.0,
 141.0,
 419.0,
 256.0,
 380.0,
 270.0,
 45.0,
 176.0,
 379.0,
 210.0,
 264.0,
 402.0,
 53.0,
 252.0,
 82.0,
 5.0,
 362.0,
 238.0,
 318.0,
 280.0,
 140.0,
 107.0,
 131.0,
 117.0,
 84.0,
 291.0,
 132.0,
 327.0,
 325.0,
 331.0,
 378.0,
 283.0,
 114.0,
 77.0,
 376.0,
 211.0,
 112.0,
 296.0,
 193.0,
 100.0,
 248.0,
 231.0,
 399.

In [221]:
df.loc[(df["availability 365"] < 0) | (df["availability 365"] > 365), "availability 365"] = np.nan
df = df.dropna(subset=['availability 365'])

In [222]:
df['minimum nights'].unique().tolist()

[10.0,
 30.0,
 45.0,
 2.0,
 1.0,
 5.0,
 4.0,
 3.0,
 7.0,
 8.0,
 15.0,
 47.0,
 81.0,
 144.0,
 371.0,
 268.0,
 -10.0,
 189.0,
 340.0,
 128.0,
 350.0,
 166.0,
 142.0,
 14.0,
 200.0,
 11.0,
 452.0,
 13.0,
 21.0,
 27.0,
 18.0,
 6.0,
 -1.0,
 -12.0,
 31.0,
 398.0,
 28.0,
 -3.0,
 90.0,
 120.0,
 365.0,
 180.0,
 122.0,
 240.0,
 29.0,
 20.0,
 88.0,
 115.0,
 150.0,
 9.0,
 370.0,
 19.0,
 60.0,
 16.0,
 26.0,
 80.0,
 25.0,
 265.0,
 300.0,
 17.0,
 40.0,
 59.0,
 360.0,
 56.0,
 50.0,
 12.0,
 70.0,
 24.0,
 32.0,
 110.0,
 270.0,
 75.0,
 62.0,
 23.0,
 1250.0,
 74.0,
 22.0,
 198.0,
 100.0,
 500.0,
 43.0,
 35.0,
 91.0,
 480.0,
 53.0,
 99.0,
 160.0,
 999.0,
 186.0,
 366.0,
 68.0,
 93.0,
 87.0,
 183.0,
 299.0,
 55.0,
 175.0,
 98.0,
 133.0,
 181.0,
 354.0,
 42.0,
 33.0,
 37.0,
 225.0,
 400.0,
 105.0,
 184.0,
 153.0,
 134.0,
 222.0,
 58.0,
 210.0,
 275.0,
 182.0,
 44.0,
 954.0,
 5645.0,
 2645.0,
 -1223.0,
 -365.0,
 -200.0,
 114.0,
 85.0,
 36.0,
 48.0,
 155.0,
 307.0,
 129.0,
 145.0,
 64.0,
 364.0,
 333.0,
 119.0

In [223]:
df.loc[(df["minimum nights"] < 1) | (df["minimum nights"] > 365), "minimum nights"] = np.nan
df = df.dropna(subset=['minimum nights'])

In [224]:
df.isna().sum()

id                                0
name                              0
host id                           0
host identity verified            0
host name                         0
neighbourhood group               0
neighbourhood                     0
lat                               0
long                              0
country                           0
country code                      0
instant bookable                  0
cancellation policy               0
room type                         0
construction year                 0
price                             0
service fee                       0
minimum nights                    0
number of reviews                 0
last review                       0
reviews per month                 0
review rate number                0
calculated host listings count    0
availability 365                  0
dtype: int64

In [225]:
info = pd.DataFrame({
    "col": df.columns,
    "dtype": df.dtypes.astype(str).values,
    "missing": df.isna().sum().values
})
info["missing_%"] = (info["missing"]/len(df)*100).round(2)
info.sort_values("missing_%", ascending=False)


,col,dtype,missing,missing_%
0,id,int64,0,0.0
1,name,object,0,0.0
22,calculated host listings count,float64,0,0.0
21,review rate number,float64,0,0.0
20,reviews per month,float64,0,0.0
19,last review,datetime64[ns],0,0.0
18,number of reviews,float64,0,0.0
17,minimum nights,float64,0,0.0
16,service fee,int64,0,0.0
15,price,float64,0,0.0


# Data Visualization

In this section, we create visualizations to explore patterns and relationships in the Airbnb dataset.

## 📊 CHART 1 — Price Distribution

**Objective:** Visualize the distribution of prices and identify outliers in the dataset.

In [226]:
plt.figure(figsize=(9,5))

sns.set_style("whitegrid")
sns.histplot(df['price'], kde=True, bins=20, )
plt.xlabel("Price")
plt.ylabel("Count")
plt.title("Price Distribution")

plt.savefig('images/price_distribution_histogram.png', dpi=300, bbox_inches='tight')
plt.close()


## 📊 CHART 2 — Price by Room Type

**Objective:** Examine the impact of room type on pricing to understand how different accommodation types are priced.

In [227]:
price_room = df.groupby("room type")["price"].median().sort_values(ascending=False)

plt.figure(figsize=(7,4))
plt.bar(price_room.index, price_room.values)
plt.xlabel("Room Type")
plt.ylabel("Median Price")
plt.title("Median Price by Room Type")
plt.grid(True, axis="y")
plt.savefig('images/median_price_by_room_type.png', dpi=300, bbox_inches='tight')
plt.close()


## 📊 CHART 3 — Room Type Distribution

**Objective:** Understand which room types are most common in the dataset and their relative proportions.

In [228]:
room_counts = df["room type"].value_counts().sort_values(ascending=False)

display(room_counts)
plt.figure(figsize=(8, 5))
plt.bar(room_counts.index, room_counts.values)
plt.xlabel("Room Type")
plt.ylabel("Count")
plt.title("Distribution of Room Types")
plt.xticks(rotation=45, ha='right')
plt.grid(True, axis="y", alpha=0.3)

plt.tight_layout()
plt.savefig('images/room_type_distribution.png', dpi=300, bbox_inches='tight')
plt.close()


room type
Entire home/apt    49918
Private room       43323
Shared room         2074
Hotel room           108
Name: count, dtype: int64

## 📊 CHART 4 — Price by Neighbourhood Group

**Objective:** Analyze the impact of location (neighbourhood group) on pricing across different areas of the city.

In [229]:
price_ng = df.groupby("neighbourhood group")["price"].median().sort_values(ascending=False)

plt.figure(figsize=(8,4))
plt.bar(price_ng.index, price_ng.values)
plt.xticks(rotation=30, ha="right")
plt.xlabel("Neighbourhood Group")
plt.ylabel("Median Price")
plt.title("Median Price by Neighbourhood Group")

plt.grid(True)
plt.savefig('images/median_price_by_neighbourhood.png', dpi=300, bbox_inches='tight')
plt.close()


## 📊 CHART 5 — Instant Bookable vs Price

**Objective:** Determine whether instant booking availability creates a price difference and affects listing pricing strategy.

In [230]:
ib_price = df.groupby("instant bookable")["price"].median()

plt.figure(figsize=(6,4))
plt.bar(ib_price.index.astype(str), ib_price.values)
plt.xlabel("Instant Bookable")
plt.ylabel("Median Price")
plt.title("Price vs Instant Bookable")
plt.grid(True, axis="y")
plt.savefig('images/price_vs_instant_bookable.png', dpi=300, bbox_inches='tight')
plt.close()


In [231]:
df['minimum nights'].unique().sum()

12906.0

## 📊 CHART 6 — Reviews per Month (Demand Indicator)

**Objective:** Identify which neighbourhoods are more active by analyzing review frequency as a proxy for demand.

In [232]:
rpm_ng = df.groupby("neighbourhood group")["reviews per month"].median().sort_values(ascending=False)

plt.figure(figsize=(8,4))
plt.bar(rpm_ng.index, rpm_ng.values)
plt.xticks(rotation=30, ha="right")
plt.xlabel("Neighbourhood Group")
plt.ylabel("Median Reviews per Month")
plt.title("Demand by Neighbourhood Group")
plt.grid(True, axis="y")
plt.savefig('images/demand_by_neighbourhood.png', dpi=300, bbox_inches='tight')
plt.close()


In [233]:
df['calculated host listings count'].unique()

array([  6.,   2.,   1.,   4.,   3.,   5.,  28.,   8.,  11.,   9.,   7.,
        52.,  15.,  19.,  39.,  10.,  13.,  26.,  29.,  12.,  18.,  21.,
        96.,  14.,  34.,  43., 121.,  37.,  49.,  31.,  91.,  16.,  87.,
        33.,  23.,  50.,  20.,  25., 232.,  17.,  47., 103.,  65.,  30.,
        27., 327.,  32.,  53., 208., 171.,  44.,  48., 108.,  51., 161.,
       152.,  83.,  24.,  22., 126.,  54.,  66.,  59.,  62., 102., 186.,
       125.,  55.,  63.,  67.,  58.,  56.,  80.,  46., 218.,  86., 332.])

## 📊 CHART 7 — Multi-Host Effect

**Objective:** Examine the pricing strategy of hosts with multiple listings to understand their market approach.

In [234]:
host_price = df.groupby("calculated host listings count")["price"].median().head(10).sort_values(ascending=False)

plt.figure(figsize=(7,4))
plt.bar(host_price.index, host_price.values)
plt.xlabel("Host Listing Count")
plt.ylabel("Median Price")
plt.title("Multi-host Pricing Behavior")
plt.grid(True, axis="y")
plt.savefig('images/multihost_pricing_behavior.png', dpi=300, bbox_inches='tight')
plt.close()


In [235]:
df

,id,name,host id,host identity verified,host name,neighbourhood group,neighbourhood,lat,long,country,...,construction year,price,service fee,minimum nights,number of reviews,last review,reviews per month,review rate number,calculated host listings count,availability 365
0,1001254,Clean & quiet apt home by the park,80014485718,unconfirmed,Madaline,Brooklyn,Kensington,40.64749,-73.97237,United States,...,2020.0,966.0,193,10.0,9.0,2021-10-19,0.21,4.0,6.0,286.0
1,1002102,Skylit Midtown Castle,52335172823,verified,Jenna,Manhattan,Midtown,40.75362,-73.98377,United States,...,2007.0,142.0,28,30.0,45.0,2022-05-21,0.38,4.0,2.0,228.0
4,1003689,Entire Apt: Spacious Studio/Loft by central park,92037596077,verified,Lyndon,Manhattan,East Harlem,40.79851,-73.94399,United States,...,2009.0,204.0,41,10.0,9.0,2018-11-19,0.10,3.0,1.0,289.0
7,1005202,BlissArtsSpace!,90821839709,unconfirmed,Emma,Brooklyn,Bedford-Stuyvesant,40.68688,-73.95596,United States,...,2009.0,1060.0,212,45.0,49.0,2017-10-05,0.40,5.0,1.0,219.0
8,1005754,Large Furnished Room Near B'way,79384379533,verified,Evelyn,Manhattan,Hell's Kitchen,40.76489,-73.98493,United States,...,2005.0,1018.0,204,2.0,430.0,2019-06-24,3.47,3.0,1.0,180.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
102029,57351953,Cozy room in a 2B with backyard access,5155155913,unconfirmed,Cristina,Queens,Astoria,40.76987,-73.92679,United States,...,2018.0,822.0,164,1.0,9.0,2019-04-25,0.91,3.0,1.0,0.0
102030,57352505,An East Williamsburg Delight,71564706996,verified,Petra,Brooklyn,Bedford-Stuyvesant,40.68335,-73.91222,United States,...,2015.0,455.0,91,4.0,36.0,2018-06-27,1.28,1.0,1.0,0.0
102031,57353057,Cozy bedroom in williamsburg,83101851929,verified,Russell,Brooklyn,Williamsburg,40.70870,-73.96710,United States,...,2020.0,1078.0,216,1.0,58.0,2019-06-10,2.08,5.0,2.0,120.0
102032,57353610,Spacious Room w/ 2 Bed,69037598918,unconfirmed,Christine,Queens,Bayside,40.75047,-73.75349,United States,...,2007.0,103.0,21,7.0,40.0,2019-06-23,1.51,1.0,5.0,148.0


# Exploratory Data Analysis Summary

This study analyzed the Airbnb Open Data to examine pricing, demand, and market structure. The analysis revealed that **room type** and **location** have the most significant impact on pricing.

## Key Findings:

- **Instant Bookable Listings:** Tend to be priced higher, suggesting a premium for convenience
- **Availability:** Listings with lower availability tend to be more expensive, indicating higher demand
- **Central Locations:** Show both higher prices and higher demand (more reviews)
- **Multi-Host Strategy:** Users with multiple listings employ more competitive pricing strategies

## Implications:

These findings provide a strong foundation for:
- Building price prediction models
- Conducting market analysis
- Understanding competitive dynamics in the short-term rental market

The insights can help both hosts optimize their pricing strategies and guests make informed booking decisions.